examples/fungible-allowlist/src/contract.rs

```diff
@@ -6,66 +6,66 @@
 //! accounts.
 
 use soroban_sdk::{contract, contractimpl, symbol_short, Address, Env, String};
-use stellar_access::access_control::{self as access_control, AccessControl};
-use stellar_macros::{default_impl, only_role};
+use stellar_access::{AccessControl, AccessController};
+use stellar_macros::only_role;
 use stellar_tokens::fungible::{
     allowlist::{AllowList, FungibleAllowList},
     burnable::FungibleBurnable,
-    Base, FungibleToken,
+    FungibleToken,
 };
 
 #[contract]
 pub struct ExampleContract;
 
 #[contractimpl]
 impl ExampleContract {
-    pub fn __constructor(e: &Env, admin: Address, manager: Address, initial_supply: i128) {
-        Base::set_metadata(
+    pub fn __constructor(e: &Env, admin: &Address, manager: &Address, initial_supply: i128) {
+        Self::set_metadata(
             e,
             18,
             String::from_str(e, "AllowList Token"),
             String::from_str(e, "ALT"),
         );
 
-        access_control::set_admin(e, &admin);
+        Self::init_admin(e, admin);
 
         // create a role "manager" and grant it to `manager`
-        access_control::grant_role_no_auth(e, &admin, &manager, &symbol_short!("manager"));
+        Self::grant_role_no_auth(e, admin, manager, &symbol_short!("manager"));
 
         // Allow the admin to transfer tokens
-        AllowList::allow_user(e, &admin);
+        Self::allow_user_no_auth(e, admin);
 
         // Mint initial supply to the admin
-        Base::mint(e, &admin, initial_supply);
+        Self::internal_mint(e, admin, initial_supply);
```

Comment:
Me:
> `internal_mint` is a new addition?

Willem:
> `mint` was public but added an `internal_mint` to the `FT` trait with `#[internal]`. This way if you you wanted to swap of the default implementation it would be guaranteed to have it.

Reply:
I'm not sure I fully get it, could you elaborate on that a bit more @willemneal ?